subsys/bluetooth/services/ras/rreq/ras_rreq.c

```diff
@@ -0,0 +1,714 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/uuid.h>
+#include <zephyr/bluetooth/gatt.h>
+#include <zephyr/bluetooth/ead.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/check.h>
+#include <zephyr/net_buf.h>
+#include <bluetooth/services/ras.h>
+#include <bluetooth/gatt_dm.h>
+#include <stdint.h>
+#include <errno.h>
+
+#include "../ras_internal.h"
+#include "zephyr/sys/__assert.h"
+
+LOG_MODULE_REGISTER(ras_rreq, CONFIG_BT_RAS_RREQ_LOG_LEVEL);
+
+enum bt_ras_rreq_cp_state {
+	BT_RAS_RREQ_CP_STATE_NONE,
+	BT_RAS_RREQ_CP_STATE_GET_RD_WRITTEN,
+	BT_RAS_RREQ_CP_STATE_ACK_RD_WRITTEN,
+};
+
+struct bt_ras_rreq_cp {
+	struct bt_gatt_subscribe_params subscribe_params;
+	enum bt_ras_rreq_cp_state state;
+};
+struct bt_ras_on_demand_rd {
+	struct bt_gatt_subscribe_params subscribe_params;
+	struct net_buf_simple *ranging_data_out;
+	bt_ras_rreq_ranging_data_get_complete_t cb;
+	uint16_t counter_in_progress;
+	uint8_t next_expected_segment_counter;
+	bool data_get_in_progress;
+	bool last_segment_received;
+	bool error_with_data_receive;
```

Comment:
```suggestion
	int error_status;
```

or similiar would make it possible to forward a more specific error code to the user callback here https://github.com/nrfconnect/sdk-nrf/pull/18737/files#diff-3f67cd569279ac64cbc075871eff1096b3735294104897f17ca416a75866ace7R137 based on what error condition was encountered

Reply:
Snap https://github.com/nrfconnect/sdk-nrf/pull/18737#discussion_r1834536849 😆

Reply:
great minds think alike :P ?